frontend/src/contexts/wishlist-context.tsx

```diff
@@ -0,0 +1,79 @@
+import { createContext, ReactNode, useState } from 'react';
+
+import { WishlistInput } from '@autoline/shared/common/types/types';
+import { HeartIcon } from '@components/common/icons/icons';
+import { Notification } from '@components/common/notification/notification';
+import {
+  useCreateWishlistMutation,
+  useDeleteWishlistMutation,
+  useGetWishlistStatusQuery,
+} from '@store/queries/preferences/wishlist';
+
+type WishlistContextType = {
+  likedCars: string[] | undefined;
+  likeClick: (data: WishlistInput) => void;
+};
+
+const WishlistContext = createContext<WishlistContextType>({
+  likedCars: undefined,
+  likeClick: () => undefined,
+});
+
+const WishlistContextProvider: React.FC<{ children: ReactNode }> = ({
+  children,
+}) => {
+  const [isMessageHidden, setIsMessageHidden] = useState<boolean>(true);
+  const [deletedCar, setDeletedCar] = useState<WishlistInput>();
+  const { data: likedCars } = useGetWishlistStatusQuery();
+
+  const [createWishlist] = useCreateWishlistMutation();
+  const [deleteWishlist] = useDeleteWishlistMutation();
+
+  const handleCreateWishlist = async (data: WishlistInput): Promise<void> => {
+    await createWishlist(data);
+  };
+
+  const handleDeleteWishlist = async (data: WishlistInput): Promise<void> => {
+    await deleteWishlist(data);
+    setIsMessageHidden(false);
+    setDeletedCar(data);
+  };
```

Comment:
Why do we add a car to `deleted` only after the deletion will be completed? The optimistic updates didn't work out?